### PR TITLE
Document that QoiImagePlugin uses Python for decoding

### LINKED ISSUE #7922: Reading Quite OK Image (QOI) files is slow compared to PNG and qoi
### What did you do?
```python
image = Image.open(image_filepath)
```
while using PyTorch data loader with 8 workers.

### What did you expect to happen?
Fast

### What actually happened?
Slow

### What are your OS, Python and Pillow versions?

* OS: Linux
* Python: 3.10.13
* Pillow: 10.2.0

Reading PNG images with Pillow is faster.

For me, current workaround is to use https://github.com/kodonnell/qoi from PyPI (`qoi` package). It is faster when I switched to it. To compare:
```python
image = qoi.read(image_filepath)
```

Image sizes are regular, like 800x480, 1280x600.

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #7922

https://github.com/python-pillow/Pillow/issues/7922#issuecomment-2033336635
> That said, Pillow's processing of QOI files in Python, not C, could probably be clarified here: https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#qoi.

### CODE REVIEW

**Comment by hugovk** [comment]
Shall we also mention https://github.com/kodonnell/qoi or https://pypi.org/project/qoi/? It's not a Pillow plugin but can be a useful alternative.

File: docs/handbook/image-file-formats.rst [hunk 1]
```diff
@@ -1488,7 +1488,9 @@ QOI
 
 .. versionadded:: 9.5.0
 
-Pillow identifies and reads images in Quite OK Image format.
+Pillow reads images in Quite OK Image format, using a Python decoder. If you wish to
+write code specifically for this format, https://pypi.org/project/qoi is an alternative
+library that interfaces with NumPy.
```
Comment (hugovk) [inline docs/handbook/image-file-formats.rst:1493]: Mention C somehow, so they know it's faster?

```suggestion
library written in C that interfaces with NumPy.
```

Comment (radarhere) [reply docs/handbook/image-file-formats.rst]: "written in C" sounds to me like the entire library is in C. How about "that uses C to decode the image"?

Comment (hugovk) [reply docs/handbook/image-file-formats.rst]: Yeah, something like that's fine too.

Comment (radarhere) [reply docs/handbook/image-file-formats.rst]: Cool, I've pushed a commit for that.

File: docs/handbook/image-file-formats.rst [hunk 2]
```diff
@@ -1488,7 +1488,9 @@ QOI
 
 .. versionadded:: 9.5.0
 
-Pillow identifies and reads images in Quite OK Image format.
+Pillow reads images in Quite OK Image format, using a Python decoder. If you wish to
+write code specifically for this format, https://pypi.org/project/qoi is an alternative
```
Comment (hugovk) [inline docs/handbook/image-file-formats.rst:1492]: Just remembered, we have an extlink role for this:

```suggestion
write code specifically for this format, :pypi:`qoi` is an alternative
```

**Review by hugovk — APPROVED** [review]

File: docs/handbook/image-file-formats.rst [hunk 3]
```diff
@@ -1488,7 +1488,9 @@ QOI
 
 .. versionadded:: 9.5.0
 
-Pillow identifies and reads images in Quite OK Image format.
+Pillow reads images in Quite OK Image format, using a Python decoder. If you wish to
+write code specifically for this format, :pypi:`qoi` is an alternative
+library that uses C to decode the image, and interfaces with NumPy.
```
Comment (nulano) [inline docs/handbook/image-file-formats.rst:1493]: Feel free to ignore, but this reads better to me.
```suggestion
Pillow reads images in Quite OK Image format using a Python decoder. If you wish to
write code specifically for this format, :pypi:`qoi` is an alternative
library that uses C to decode the image and interfaces with NumPy.
```

**Review by nulano — APPROVED** [review]

**Pull request merged**